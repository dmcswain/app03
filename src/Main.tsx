import { useMemo } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { CssBaseline, ThemeProvider, useMediaQuery } from '@mui/material';
import getTheme from './theme';
import StoreProvier from 'store/Provider';

interface Props {
   children: React.ReactNode;
}

const Main: React.FC<Props> = ({ children }) => {
   const isDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
   const theme = useMemo(() => getTheme(isDarkMode), [isDarkMode]);

   return (
      <ThemeProvider theme={theme}>
         <BrowserRouter>
            <StoreProvier>
               <CssBaseline />
               {children}
            </StoreProvier>
         </BrowserRouter>
      </ThemeProvider>
   );
};

export default Main;
