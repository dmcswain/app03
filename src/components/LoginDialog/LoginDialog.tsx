import { useCallback, useReducer } from 'react';
import {
   Box,
   Button,
   Dialog,
   DialogContent,
   DialogTitle,
   TextField,
} from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { useDispatch } from 'store/Provider';
import { addUser } from 'db';

export interface LoginDialogProps {}

const LoginDialog: React.FC<LoginDialogProps> = () => {
   const navigate = useNavigate();
   const dispatch = useDispatch();
   const [open, toggleDialog] = useReducer(state => !state, true);

   const handleClose = useCallback(() => {
      toggleDialog();
      navigate(-1);
   }, [navigate]);

   const handleSubmit = useCallback(
      (event: React.ChangeEvent<HTMLFormElement>) => {
         event.preventDefault();

         const formData = new FormData(event.currentTarget);
         const formEntries = Object.fromEntries(
            formData.entries() as IterableIterator<[string, string]>
         );

         const currentUser = addUser(formEntries as unknown as I.User);

         dispatch({ type: 'login', payload: currentUser });
         toggleDialog();
         navigate('/', { replace: true });
      },
      [dispatch, navigate]
   );

   return (
      <Dialog maxWidth='sm' fullWidth open={open} onClose={handleClose}>
         <DialogTitle>Login</DialogTitle>

         <DialogContent>
            <Box
               component='form'
               onSubmit={handleSubmit}
               sx={{
                  display: 'flex',
                  flexDirection: 'column',
                  flexShrink: 0,
                  gap: theme => theme.spacing(1),
               }}
            >
               <TextField
                  sx={{ my: 1 }}
                  name='username'
                  placeholder='Username'
                  label='Username'
                  required
                  fullWidth
               />

               <TextField
                  sx={{ my: 1 }}
                  name='password'
                  type='password'
                  placeholder='Password'
                  label='Password'
                  required
                  fullWidth
               />

               <Button
                  type='submit'
                  variant='contained'
                  color='primary'
                  size='large'
               >
                  Login
               </Button>
            </Box>
         </DialogContent>
      </Dialog>
   );
};

export default LoginDialog;
