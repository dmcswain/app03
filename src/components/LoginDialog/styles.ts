import { DialogContent, styled } from '@mui/material';

export const DialogContentStyle = styled(DialogContent)(({ theme }) => ({
   display: 'flex',
   flexDirection: 'column',
   flexShrink: 0,
   gap: theme.spacing(1),
   '.field': {
      marginTop: theme.spacing(1),
      marginBottom: theme.spacing(1),
   },
}));
