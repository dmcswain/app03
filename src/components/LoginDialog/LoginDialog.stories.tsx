import { Story, ComponentMeta } from '@storybook/react';
import Main from 'Main';
import LoginDialog from './LoginDialog';

export const Default = () => <LoginDialog />;

Default.decorators = [
   (Story: Story) => (
      <Main>
         <Story />
      </Main>
   ),
];

export default {
   component: LoginDialog,
} as ComponentMeta<typeof LoginDialog>;
