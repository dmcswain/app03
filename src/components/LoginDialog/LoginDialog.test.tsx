import { render, screen } from '@testing-library/react';
import Main from 'Main';
import LoginDialog, { LoginDialogProps } from './LoginDialog';

describe('LoginDialog', () => {
   const defaultProps: LoginDialogProps = {};

   it('should render', () => {
      const wrapper = ({ children }: { children: React.ReactNode }) => (
         <Main>{children}</Main>
      );

      const props = { ...defaultProps };
      const { asFragment } = render(<LoginDialog {...props} />, { wrapper });

      expect(asFragment()).toMatchSnapshot();
      expect(screen.getAllByText('Login')).toBeTruthy();
   });
});
