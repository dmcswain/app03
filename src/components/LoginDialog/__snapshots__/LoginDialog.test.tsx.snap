// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDialog should render 1`] = `
<DocumentFragment>
  ;
</DocumentFragment>
`;
